urdf_usd_converter/_impl/conversion_collada.py

```diff
@@ -0,0 +1,192 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 The Newton Developers
+# SPDX-License-Identifier: Apache-2.0
+import pathlib
+
+import collada
+import numpy as np
+import usdex.core
+from pxr import Gf, Tf, Usd, UsdGeom, Vt
+
+from .data import ConversionData
+from .numpy import convert_face_indices_array, convert_matrix4d, convert_vec2f_array, convert_vec3f_array
+
+__all__ = ["ConversionCollada"]
+
+
+class ConversionCollada:
+    def __init__(self, input_path: pathlib.Path):
+        # Collada mesh IDs and USD.prim dictionary.
+        self.meshes: dict[str, Usd.Prim] = {}
+
+        try:
+            self.collada = collada.Collada(str(input_path))
+
+            # The default unit for the scene is meters (= 1.0).
+            self.unit_meter = self.collada.assetInfo.unitmeter if self.collada.assetInfo.unitmeter is not None else 1.0
+
+        except Exception as e:
+            self.collada = None
+            Tf.Warn(f'Invalid input_path: "{input_path!s}" could not be parsed. {e}')
+
+    def convert(self, prim: Usd.Prim, data: ConversionData) -> Usd.Prim | None:
+        """
+        Trace the hierarchical structure and place the geometry.
```

Comment:
Just commenting here to have an open thread.

> Below is a conceptual diagram of dae.

I don't think we should be doing this. It would require a full Conceptual Mapping of DAE to USD to get this right. I think we should limit scope as tightly as possible. At that point, there should probably be a separate project `collada_usd_converter` and that is out-of-scope for Newton.

Our SOL would be only supporting single-scene-single-mesh DAE files.

Please explore the `urdf_files_dataset` & https://github.com/newton-physics/newton-assets & determine if that is viable
- How many assets use multiple scenes?
- How many assets use multiple meshes?
- How many assets use non-mesh geometries?
- How many assets use references to meshes across geometries?